server/src/main/java/org/candlepin/controller/ProductManager.java

```diff
@@ -629,18 +704,57 @@ private Product applyProductChanges(Product entity, ProductDTO update, Owner own
     private Product applyProvidedProductChanges(Product entity, ProductDTO update, Owner owner) {
 
         if (update.getProvidedProducts() != null && !update.getProvidedProducts().isEmpty()) {
-            entity.getProvidedProducts().clear();
+            Set<Product> product = new HashSet<>();
 
             for (ProductDTO providedProductDTO : update.getProvidedProducts()) {
                 if (providedProductDTO != null && providedProductDTO.getId() != null) {
                     Product newProd = this.ownerProductCurator.getProductById(owner,
                         providedProductDTO.getId());
 
                     if (newProd != null) {
-                        entity.addProvidedProduct(newProd);
+                        product.add(newProd);
                     }
                 }
             }
+
+            entity.setProvidedProducts(product);
+        }
+
+        return entity;
+    }
+
+    /**
+     * Applies the changes related to provided products from the given productInfo to the specified entity.
+     *
+     * @param entity
+     *  The entity to modify.
+     *
+     * @param update
+     *  The productInfo interface.
+     *
+     * @param owner
+     *  An owner to use for resolving entity references.
+     *
+     * @return
+     *  The updated product entity.
+     */
+    private Product applyProvidedProductChanges(Product entity, ProductInfo update, Owner owner) {
+
+        if (update.getProvidedProducts() != null && !update.getProvidedProducts().isEmpty()) {
+            Set<Product> product = new HashSet<>();
+
+            for (ProductInfo pInfo : update.getProvidedProducts()) {
+                if (pInfo != null && pInfo.getId() != null) {
+                    Product newProd = this.ownerProductCurator.getProductById(owner,
```

Comment:
Keeping this block of code as a part of workaround, since Product import will 
be completely refactored once the design is finalized, this will eventually get removed.